src/main/java/javaslang/algebra/Foldable.java

```diff
@@ -126,4 +129,63 @@ default T foldRight(Monoid<? extends T> monoid) {
      */
     <U> U foldRight(U zero, BiFunction<? super T, ? super U, ? extends U> combine);
 
+    /**
+     * Accumulates the elements of this Foldable by successively calling the given operation {@code op}.
+     * The order of element iteration is undetermined.
+     *
+     * @param op A BiFunction of type T
+     * @return the reduced value.
+     * @throws NoSuchElementException if this is empty
+     * @throws NullPointerException   if {@code op} is null
+     */
+    T reduce(BiFunction<? super T, ? super T, ? extends T> op);
```

Comment:
Let's take the default impl from Traversable here, then we can remove the impls in Value, Iterator and Traversable